clang/test/SemaSYCL/CudaCompat/delayed-diags.cu

```diff
@@ -0,0 +1,43 @@
+// RUN: %clang_cc1 %s -fsycl-is-host -fsycl-cuda-compatibility \
+// RUN:   -internal-isystem %S/../../SemaCUDA/Inputs \
+// RUN:   -internal-isystem %S/../Inputs \
+// RUN:   -fsycl-targets=nvptx64-nvidia-cuda -triple x86_64-unknown-linux \
+// RUN:   -o %t -verify
+// RUN: %clang_cc1 %s -fsycl-is-device -fsycl-cuda-compatibility \
+// RUN:   -internal-isystem %S/../../SemaCUDA/Inputs \
+// RUN:   -internal-isystem %S/../Inputs \
+// RUN:   -fsycl-targets=nvptx64-nvidia-cuda -triple x86_64-unknown-linux\
+// RUN:   -o %t -verify
+
+// Check that delayed diagnostics are done
+// according to the SYCL logic and not the CUDA one.
+// fsycl-cuda-compatibility doesn't enable the CUDA device mode,
+// this leaves the diagnostic logic for CUDA it is processing the host side
+// and triggers SYCL's delayed diagnostics.
```

Comment:
Grammar seems wrong; "this leaves the diagnostic logic for CUDA it is processing the host side".